instrumentation/internal/internal-class-loader/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/internal/classloader/ClassLoadingTest.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.internal.classloader;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.net.URL;
+import java.net.URLClassLoader;
+import org.junit.jupiter.api.Test;
+
+class ClassLoadingTest {
+
+  @Test
+  void testDelegatesToBootstrapClassLoaderForAgentClasses() {
+    ClassLoader classLoader = new NonDelegatingUrlClassLoader();
+    Class<?> clazz = null;
```

Comment:
I'd add `throws ClassNotFoundException` to the method as it is done in osgi and tomcat test